src/Admin/TemplateRegistryInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+interface TemplateRegistryInterface
+{
+    /**
+     * @return array
+     */
+    public function getTemplates();
+
+    public function setTemplates(array $templates);
+
+    /**
+     * @param string $name
+     *
+     * @return string
+     */
+    public function getTemplate($name);
+
+    /**
+     * @param string $name
+     * @param string $template
+     */
+    public function setTemplate($name, $template);
```

Comment:
I think we could move the setters to a `MutableTemplateRegistryInterface`, and maybe remove them on master? And use a constructor instead?

Reply:
I'm now using the constructor.

Would adding the new interface, plus the future servicelocator, justify a separate namespace, as we then have 4 classes/interfaces that belong together?

Reply:
I think it would :+1: `Templating`?

Reply:
Inside `Admin`?

Reply:
For now it's a component, but later it will be independent, so I'd rather say outside, but that's just my opinion.

Reply:
If I split up into a `TemplateRegistryInterface` and a `MutableTemplateRegistryInterface`, how should I do the typehinting? And should one interface extend the other?

Reply:
I think `MutableTemplateRegistryInterface` should extend `TemplateRegistryInterface`, and the type hinting should be done depending on what the consuming class requires. For instance, the twig extensions will not need to set any template, will they?

Reply:
Alright, thanks. :-)